var/spack/repos/builtin/packages/r/package.py

```diff
@@ -127,6 +129,24 @@ class R(AutotoolsPackage):
 
     build_directory = "spack-build"
 
+    @classmethod
+    def determine_version(cls, exe):
+        output = Executable(exe)("--version", output=str, error=str)
+        # R version 4.3.3 (2024-02-29) -- "Angel Food Cake"
```

Comment:
If you need it, you can add integration tests for detection. Look at the `detection_test.yaml` e.g. under the `gcc` folder